searchcore/src/tests/proton/attribute/attribute_test.cpp

```diff
@@ -773,6 +779,95 @@ TEST_F("require that AttributeWriter::forceCommit() clears search cache in impor
     EXPECT_EQUAL(0u, f._m->getImportedAttributes()->get("imported_b")->getSearchCache()->size());
 }
 
+struct StructFixtureBase : public Fixture
+{
+    DocumentType _type;
+    const Field _valueField;
+    StructDataType _structFieldType;
+
+    StructFixtureBase()
+        : Fixture(),
+          _type("test"),
+          _valueField("value", 2, *DataType::INT, true),
+          _structFieldType("struct")
+    {
+        addAttribute({"value", AVConfig(AVBasicType::INT32, AVCollectionType::SINGLE)}, createSerialNum);
+        _type.addField(_valueField);
+        _structFieldType.addField(_valueField);
+    }
+
+    std::unique_ptr<StructFieldValue>
+    makeStruct()
+    {
+        return std::make_unique<StructFieldValue>(_structFieldType);
+    }
+
+    std::unique_ptr<StructFieldValue>
+    makeStruct(const int32_t value)
+    {
+        auto ret = makeStruct();
+        ret->setValue(_valueField, IntFieldValue(value));
+        return ret;
+    }
+
+    std::unique_ptr<Document>
+    makeDoc()
+    {
+        return std::make_unique<Document>(_type, DocumentId("id::test::1"));
+    }
+};
+
+
+struct StructArrayFixture : public StructFixtureBase
+{
+    using StructFixtureBase::makeDoc;
+    const ArrayDataType _structArrayFieldType;
+    const Field _structArrayField;
+
+    StructArrayFixture()
+        : StructFixtureBase(),
+          _structArrayFieldType(_structFieldType),
+          _structArrayField("s", _structArrayFieldType, true)
```

Comment:
I was a bit confused with the name of this field, and the variables 'svalues' used further down. Consider rename 's' -> 'array' and 'svalues' -> 'arrayValues'.